syntax = "proto2";

package NAlice.NMusicWebSearch.NListeningCounters;

message TUserListening {
    optional string Puid = 1;
    optional string ReqId = 2;
    optional double Timespent = 3;
    optional int64 Timestamp = 4;
    optional string Url = 5;
}

message TFullState {
    optional string Puid = 1;
    optional string StateId = 2;
    optional string State = 3;
    optional string StateDebug = 4;
}

message TCompactState {
    optional uint64 Hash = 1;
    optional string Id = 2;
    optional string Codec = 3;
    optional string State = 4;
    optional string StateDiff = 5;
    optional string StatePatch = 6;
}
